// codeceptjs-bshelper/dist/lib/ApiHelper.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ApiHelper = void 0;
const axios_1 = __importDefault(require("axios"));
class ApiHelper {
    constructor() {
        this.axios = axios_1.default.create();
    }
    async makePostRequest(url, payload = {}, headers = {}) {
        try {
            return this.axios.post(url, payload, { headers });
        }
        catch (e) {
            throw new Error(e.message);
        }
    }
    async makePutRequest(url, payload = {}, headers = {}) {
        try {
            return this.axios.put(url, payload, { headers });
        }
        catch (e) {
            throw new Error(e.message);
        }
    }
    async makeGetRequest(url, headers = {}) {
        try {
            return this.axios.get(url, { headers });
        }
        catch (e) {
            throw new Error(e.message);
        }
    }
    async makeDeleteRequest(url, headers = {}) {
        try {
            return this.axios.delete(url, { headers });
        }
        catch (e) {
            throw new Error(e.message);
        }
    }
}
exports.ApiHelper = ApiHelper;
